### PR TITLE
Fix problem with MM hint lists sorting differently.

### PR DESCRIPTION
**Pull request by readroberts**
Even though the MM bez files are merge compatible after hinting, the T2 CharStrings built from them may not be compatible. This is because hint pairs in T2 CharStrings must be sorted, and duplicates must be eliminated. The differing values of hint pairs between the MM source fonts may cause them to be sorted and filtered differently. The fix was to define a list of the selected hint pair indices and order with the reference font, and then apply this list in the region fonts.

### CODE REVIEW

File: python/psautohint/otfFont.py [hunk 1]
```diff
@@ -634,15 +710,35 @@ def convertBezToT2(bezString):
...
+                pos = hintMask.listPos
+                hBytes = hintMask.maskByte(hhints, vhints)
+                t2List[pos] = [["hintmask"], hBytes]
+                if in_mm_hints:
+                    hm = RefHintMask(pos=pos, hbytes=hBytes)
+                    mm_hint_info.hint_masks.append(hm)
+        elif (mm_hint_info is not None):
+            # This is a MM region font: apply hint masks from reference font.
+            hm0_mask = mm_hint_info.hint_masks[0].hbytes
+            if type(t2List[0][0]) == HintMask:
```
Comment (miguelsousa) [inline python/psautohint/otfFont.py]: Use `isinstance()` instead of `type()`, as suggested by Codacy

Comment (readroberts) [reply python/psautohint/otfFont.py]: Done.

**Comment by miguelsousa** [comment]
@readroberts is it feasible to add a test that covers these changes?

**Comment by readroberts** [comment]
@msousa. There already is. The test function test_mhint.py::test_sparse_mmotf() covers this.

File: python/psautohint/otfFont.py [hunk 2]
```diff
@@ -440,17 +437,70 @@ def makeRelativeCTArgs(argList, curX, curY):
...
+    # should filter out duplicates
+
+    numHints = len(hints)
+    index_list = list(range(numHints))
+    hint_list = list(zip(hints, index_list))
+    hint_list.sort()
+    new_hints = [hint_list[i] for i in range(1, numHints)
+                 if hint_list[i][0] != hint_list[i-1][0]]
+    new_hints = [hint_list[0]] + new_hints
+    hints, hint_order = list(zip(*new_hints))
```
Comment (blueshade7) [inline python/psautohint/otfFont.py]: I believe the comment means: "calling hint_order.index(index_from_bez) will yield the hint index after sorting".
I first interpreted it backwards: "hint_order[index_from_bez] will yield the hint index after sorting".
Seems misleading; better clarify?

File: python/psautohint/otfFont.py [hunk 3]
```diff
@@ -499,33 +549,45 @@ def convertBezToT2(bezString):
             lastPathOp = token
         elif token == "rb":
             lastPathOp = token
-            try:
-                i = hhints.index(argList)
-            except ValueError:
-                i = len(hhints)
+            if in_mm_hints:
```
Comment (blueshade7) [inline python/psautohint/otfFont.py]: The same logic is repeated four times. Maybe better to factor it out as a function?

File: python/psautohint/otfFont.py (hunk 2 above)
Comment (readroberts) [reply python/psautohint/otfFont.py]: Will do this, and your other suggestion. FYI, I am working on a new commit to remove hint pairs that are incorrect in the MM hinted region fonts.

**Pull request merged**